app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,172 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from core.models import Recipe, Ingredient
+
+from recipe.serializers import RecipeSerializer, RecipeDetailSerializer
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def detail_url(recipe_id):
+    """Return recipe detail url"""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def sample_ingredient(recipe, name='love'):
```

Comment:
out of the scope of this exercise, but fyi, we use this library to generate dummy data:
https://factoryboy.readthedocs.io/en/latest/orms.html#django

Reply:
👍  this is good to know